packages/@aws-cdk/aws-ecs/lib/container-definition.ts

```diff
@@ -238,6 +238,11 @@ export class ContainerDefinition extends cdk.Construct {
    */
   public readonly ulimits = new Array<Ulimit>();
 
+  /**
+   * An array dependencies defined for container startup and shutdown.
+   */
+  public readonly containerDependencies = Array<ContainerDependency>();
```

Comment:
I am only now realizing that all of these are exposing mutable arrays to users... Not ideal. @rix0rrr what do you think?